src/composable/types/Twap.ts

```diff
@@ -19,44 +20,19 @@ export const MAX_UINT32 = BigNumber.from(2).pow(32).sub(1) // 2^32 - 1
 export const MAX_FREQUENCY = BigNumber.from(365 * 24 * 60 * 60) // 1 year
 
 // Define the ABI tuple for the TWAPData struct
-export const TWAP_DATA_ABI = [
+const TWAP_DATA_ABI = [
   'tuple(address sellToken, address buyToken, address receiver, uint256 partSellAmount, uint256 minPartLimit, uint256 t0, uint256 n, uint256 t, uint256 span, bytes32 appData)',
 ]
 
 // const DEFAULT_TOKEN_FORMATTER =  (address: string, amount: BigNumber) => string
 const DEFAULT_TOKEN_FORMATTER = (address: string, amount: BigNumber) => `${address}@${amount}`
 
 /**
- * Parameters for a TWAP order, made a little more user-friendly for SDK users.
- *
- * @see {@link TwapData} for the native struct.
- */
-export type TwapDataParams = Omit<TwapData, 'partSellAmount' | 'minPartLimit'> & {
-  // total amount of sellToken to sell across the entire TWAP
-  readonly sellAmount: BigNumber
-  // minimum amount of buyToken that must be bought across the entire TWAP
-  readonly buyAmount: BigNumber
-}
-
-/**
- * Parameters for a discrete part of a TWAP order
- */
-export type TwapPart = {
-  /**
-   * Amount of sellToken to sell in each part
-   */
-  partSellAmount: BigNumber
-
-  /**
-   * Minimum amount of buyToken that must be bought across the entire TWAP
-   */
-  minPartLimit: BigNumber
-}
-
-/**
- * Parameters for a TWAP order, as expected by the contract's `staticInput`.
+ * Base parameters for a TWAP order. Shared by:
+ *   - TwapStruct (modeling the contract's struct used for `staticInput`).
+ *   - TwapData (modeling the friendly SDK interface).
  */
-export type TwapData = {
+export type TwapDataBase = {
```

Comment:
Shared model between the struct and the data

Reply:
🚀🚀